File: auto-migrate.py
#!/usr/bin/env python3
import pexpect
import sys
import os

os.chdir('/home/ubuntu/evox-fiscal')

# Step 1: drizzle-kit generate
print("=== Running drizzle-kit generate ===")
child = pexpect.spawn('npx drizzle-kit generate --name geg_v36', timeout=120, encoding='utf-8')
child.logfile_read = sys.stdout

while True:
    try:
        idx = child.expect([
            r'create column',           # Interactive prompt asking about column
            r'renamed from another',     # Another variant
            r'Migration .* generated',   # Success
            r'No schema changes',        # No changes needed
            pexpect.EOF,
            pexpect.TIMEOUT
        ], timeout=30)
        
        if idx in [0, 1]:
            # Send Enter to select the first option (create column)
            child.sendline('')
        elif idx in [2, 3]:
            print("\n=== Generate completed successfully ===")
            child.expect(pexpect.EOF, timeout=10)
            break
        elif idx == 4:
            print("\n=== Generate process ended ===")
            break
        elif idx == 5:
            print("\n=== Timeout, sending Enter ===")
            child.sendline('')
    except pexpect.EOF:
        print("\n=== Generate process ended (EOF) ===")
        break
    except pexpect.TIMEOUT:
        print("\n=== Timeout ===")
        child.sendline('')

child.close()
print(f"Generate exit status: {child.exitstatus}")

if child.exitstatus != 0:
    print("Generate failed, exiting")
    sys.exit(1)

# Step 2: drizzle-kit migrate
print("\n=== Running drizzle-kit migrate ===")
child2 = pexpect.spawn('npx drizzle-kit migrate', timeout=120, encoding='utf-8')
child2.logfile_read = sys.stdout

try:
    child2.expect(pexpect.EOF, timeout=60)
except pexpect.TIMEOUT:
    print("\n=== Migrate timeout ===")

child2.close()
print(f"\nMigrate exit status: {child2.exitstatus}")

File: auto-migrate.mjs
import { spawn } from 'child_process';

function runCommand(cmd, args) {
  return new Promise((resolve, reject) => {
    const proc = spawn(cmd, args, {
      cwd: '/home/ubuntu/evox-fiscal',
      stdio: ['pipe', 'pipe', 'pipe'],
      env: process.env
    });

    let output = '';
    let errOutput = '';

    proc.stdout.on('data', (data) => {
      const text = data.toString();
      output += text;
      process.stdout.write(text);
      
      // Auto-select first option (create column) by sending Enter
      if (text.includes('create column') || text.includes('renamed from another column')) {
        setTimeout(() => {
          proc.stdin.write('\n');
        }, 100);
      }
    });

    proc.stderr.on('data', (data) => {
      const text = data.toString();
      errOutput += text;
      process.stderr.write(text);
    });

    proc.on('close', (code) => {
      if (code === 0) {
        resolve(output);
      } else {
        reject(new Error(`Process exited with code ${code}\n${errOutput}`));
      }
    });

    proc.on('error', reject);
  });
}

async function main() {
  console.log('=== Running drizzle-kit generate ===');
  try {
    await runCommand('npx', ['drizzle-kit', 'generate', '--name', 'geg_v36']);
    console.log('\n=== Generate completed ===');
  } catch (e) {
    console.error('Generate error:', e.message);
    process.exit(1);
  }

  console.log('\n=== Running drizzle-kit migrate ===');
  try {
    await runCommand('npx', ['drizzle-kit', 'migrate']);
    console.log('\n=== Migrate completed ===');
  } catch (e) {
    console.error('Migrate error:', e.message);
    process.exit(1);
  }
}

main();
